ucStoreAndPromoteWithDependencies.groovy

```diff
@@ -0,0 +1,361 @@
+ * - 'DEFAULT' or empty|null: if NO EXACT MATCH found for download, pick the latest version found in store (example: checking for 1.2, found 1.3 and 1.5 -> pick 1.5)
+ *
+ * Note: With either LATEST / DEFAULT, the output shows all the required dependencies
+ */
+String updateCenterStoreStrategy = "LATEST"
+/**
+ * Strategy for picking available version. When checking version available for download, the script picks the exact
+ * version if it can find it, otherwise the default behavior is to pick the latest version available. This can be
+ * controlled with this variable:
+ *
+ * - 'LATEST': pick the latest version available for download EVEN IF AN EXACT MATCH IS FOUND
+ * - 'DEFAULT' or empty|null: if NO EXACT MATCH found for download, pick the latest version found for download (example: checking for 1.2, found 1.3 and 1.5 -> pick 1.5)
+ *
+ * Note: With either LATEST / DEFAULT, the output shows all the required dependencies
+ */
+String updateCenterDownloadStrategy = "LATEST"
+
+/**
+ * Fill a map with required dependencies.
+ * @param updateCenter The update center item
+ * @param name the name of the plugin
+ * @param version the version of the plugin
+ * @param pickStrategy strategy to apply when exact version is not available
+ * @param checkedDeps the dependencies already checked
+ * @param optionalDeps the optional dependencies already checked
+ * @param requiredDeps the required dependencies (to be stored/promoted)
+ * @param indent indent for log output
+ */
+boolean fillRequiredDependencies(
+        UpdateCenter updateCenter,
+        String name,
+        String version,
+        String pickStrategy,
+        String downloadStrategy,
+        Collection<DependencyEntry> checkedDeps,
+        Map<String, VersionNumber> optionalDeps,
+        Map<String, PluginEntry> requiredDeps,
+        String indent) {
+
+    PluginData pluginData = updateCenter.getPlugin(name)
+    if(pluginData == null) {
+        println "${indent}[ERROR]Plugin [${name}] does not exists in this update center!"
+        return false
+    }
+
+    VersionNumber pluginVersionNumber = version == null ? null : new VersionNumber(version)
+    println "${indent}[${pluginData.name}:${pluginVersionNumber}]"
+
+    PluginEntry pluginStoredEntry, pluginDownloadableEntry = null
+
+    println "${indent} Checking available versions in store..."
+    pluginStoredEntry = getSuitableVersionInStore(pluginData, pluginVersionNumber, pickStrategy,"${indent} ")
+
+    if(pluginStoredEntry == null || downloadStrategy == "LATEST") {
+        println "${indent} Checking available versions for download..."
+        pluginDownloadableEntry = getSuitableVersionToDownload(pluginData, pluginVersionNumber, downloadStrategy,"${indent} ")
+    }
+
+    pluginEntry = pluginDownloadableEntry != null ? pluginDownloadableEntry : pluginStoredEntry
+
+    boolean consistent = true
+    if(pluginEntry != null) {
```

Comment:
IntelliJ IDEA complains that it cannot resolve this variable. I am guessing line 129 is missing a `def`.